src/create.ts

```diff
@@ -34,8 +36,14 @@ export async function createCommand(options: CreateOptions): Promise<void> {
   }
 
   const cwd = options.cwd ?? process.cwd();
-  const templateDir = path.join(__dirname, "..", "template");
+  const dirname = options.dirname ?? __dirname;
+  const tempDir = await mkdtemp("extract-template-");
```

Comment:
try/catch around the logic that uses the tempDir so you clean it up even if some intermediate code crashes